src/ranged.cpp

```diff
@@ -549,7 +549,7 @@ int player::fire_gun( const tripoint &target, int shots, item& gun )
             break;
         }
 
-        double dispersion = get_weapon_dispersion( &gun, true );
+        double dispersion = rng_normal( get_weapon_dispersion( gun ) ) + rng_normal( recoil_current() );
```

Comment:
> What part of, "this is working as intended, don't change it" was unclear?

I don't think the current code as a whole works well and that was the argument set out in #17643. The sum of distributions part isn't especially causative but it is a significant obstacle (see below).

> you can extract the random calls to one place, maintaining the separate sources of dispersion until it is applied.

This PR starts work on that idea - it breaks it out into player recoil and weapon parts. At `MIN_RECOIL` there isn't a significant difference (normal distribution is unbiased, original bias was 57%). At higher recoil there is (maximum tested was 72%).

The rationale is that it's impossible for `gun_engagement_range` to perform the inverse calculation for a sum of distributions due to the variable bias.

> Normal distribution is fine.

I'd agree a limited set of summed normal distributions is likely the correct model. The next steps could include working out what that set is and providing some data to check the output is sensible and writing some unit tests.